src/Modal/Modal.tsx

```diff
@@ -0,0 +1,236 @@
+ * While the default rendering behavior is often sufficient, the `renderInPortal` prop can be used
+ * to append a div to the body.
+ *
+ */
+const Modal: FC<ModalProps> = (props) => {
+    const {
+        handleClose,
+        isOpen,
+        headerText = '',
+        isCloseIconVisible = true,
+        actionButton,
+        cancelButtonText = CANCEL_BUTTON_TEXT,
+        subHeaderText = '',
+        children,
+        showTopDivider = false,
+        showBottomDivider = false,
+        dialogWidth = 'sm',
+        renderInPortal = false,
+        className
+    } = props;
+
+    const handleOnClose = () => {
+        if (handleClose) {
+            handleClose();
+        }
+    };
+
+    const [portal, setPortal] = useState<HTMLElement | null>(null);
+    const [dialogElement, setDialogElement] = useState<HTMLElement | null>(null);
+    const [update, setUpdate] = useState<number>(0);
+
+    useEffect(() => {
+        const bodyElementHTMLCollection = document.getElementsByTagName('body');
+        const bodyElement = bodyElementHTMLCollection.length > 0 ? bodyElementHTMLCollection.item(0) : null;
+        let observer: IntersectionObserver;
+        let portalElement: HTMLElement;
+
+        if (renderInPortal && bodyElement) {
+            portalElement = document.createElement('div');
+            portalElement.onclick = (e) => {
+                if (e.target !== portalElement) {
+                    return;
+                }
+                handleOnClose();
+            };
+
+            if (!portal) {
+                bodyElement.appendChild(portalElement);
+            }
+
+            if (!portal && dialogElement) {
+                observer = new IntersectionObserver(() => {
+                    setUpdate(new Date().getTime());
+                });
+
+                observer.observe(dialogElement);
+                setPortal(portalElement);
+            }
+        }
+
+        return () => {
+            if (dialogElement && observer) {
+                observer.unobserve(dialogElement);
+            }
+
+            if (portalElement && bodyElement && bodyElement.contains(portalElement)) {
+                bodyElement.removeChild(portalElement);
+            }
+        };
+    }, [dialogElement]);
+
+    useEffect(() => {
+        if (dialogElement && portal) {
+            portal.style.display = isOpen ? 'flex' : 'none';
+            portal.style.backgroundColor = 'rgb(0,0,0), rgba(0,0,0,0.4)';
+            portal.style.width = '100%';
+            portal.style.height = '100%';
+            portal.style.justifyContent = 'center';
+            portal.style.alignItems = 'center';
+            portal.style.position = 'fixed';
+            portal.style.top = '0';
+            portal.style.left = '0';
+            portal.style.zIndex = `${theme?.zIndex?.modal}`;
+        }
+    }, [update]);
+
+    const dialogueNodeMounted = (node: HTMLDivElement) => {
```

Comment:
Minor: we should probably pick a consistent spelling for `dialog` / `dialogue`.

Reply:
I vote dialog.

Reply:
Good call. Complete!